onnxruntime/core/providers/xnnpack/nn/conv.cc

```diff
@@ -64,29 +65,54 @@ Status Conv::Compute(OpKernelContext* context) const {
   if (Y->Shape().Size() == 0) {
     return Status::OK();
   }
-  pthreadpool_t t_pool = GetThreadPool();
 
-  xnn_status status = xnn_status_invalid_state;
+  pthreadpool_t threadpool = GetThreadPool();
+
+  // setup allocator/automated dellocate for workspace
+  size_t workspace_size = 0;
+  size_t workspace_alignment = 0;
```

Comment:
Wondering if we supported workspace, are we gonna support subgraph as well?

Reply:
Based on the implementation where there were errors if the workspace wasn't provided, and the tests with the PR to update the API all used a workspace, it didn't appear to be optional.

e.g. https://github.com/google/XNNPACK/blob/c7e7cde37615a81a529c326aa278bfab4cd6fe5a/src/operators/average-pooling-nhwc.c#L1157

However I just noticed I didn't actually populate the workspace pointer in Conv::Compute so maybe it's not required. Didn't really make sense to me when we're allocating it for every Compute call anyway.

IIRC using subgraphs would completely change how we used XNNPACK from creating individual kernels to building subgraphs right? if using subgraphs is a non-trivial amount of work we'd want a solid use case to justify it.